ecs-cli/modules/cli/local/network/setup.go

```diff
@@ -122,6 +132,53 @@ func createLocalNetwork(dockerClient networkCreator) {
 	logrus.Infof("Created network %s with ID %s", EcsLocalNetworkName, resp.ID)
 }
 
+func setupLocalEndpointsImage(dockerClient imagePuller) {
+	if localEndpointsImageExists(dockerClient) {
+		return
+	}
+	pullLocalEndpointsImage(dockerClient)
+}
+
+func localEndpointsImageExists(dockerClient imagePuller) bool {
+	ctx, cancel := context.WithTimeout(context.Background(), docker.TimeoutInS)
+	defer cancel()
+
+	args := filters.NewArgs(filters.Arg("reference", localEndpointsImageName))
+	imgs, err := dockerClient.ImageList(ctx, types.ImageListOptions{
+		Filters: args,
+	})
+	if err != nil {
+		logrus.Fatalf("Failed to list images with filters %v due to %v", args, err)
```

Comment:
Please correct me if I'm wrong (maybe the convention is different for CLI tools). Previously, I learned that it's better for libraries to not panic but return errors to their callers so that they can have a chance to recover, handle it using the context in which the library function call was made or log more informative messages (again using the context in which the function call was made).

Reply:
Wesley brought up a similar point in [PR-784](https://github.com/aws/amazon-ecs-cli/pull/784/#discussion_r291771434).  

For these errors I don't see any other way of handling them besides just returning an error to the user and stopping the app. I'd rather handle the error where it's created than bubbling it up if there is only one course of action to keep the code cleaner.

Let me know what you think! if you guys feel strongly about it I can bubble it up but it will result in a bunch of ``` if err != nil { return err } ``` across callers.